### PR TITLE
DIRECTOR: LINGO: Implement traceLogFile Lingo Property

### PR DESCRIPTION
**Pull request by r41k0u**
This change implements traceLogFile and the required function in g_debugger. There is a new wrapper of debugPrintf which will log the stdout and stderr output in the log file specified. Functionality has been tested and us found consistent to that of the original

### CODE REVIEW

**Review by sev- — CHANGES_REQUESTED** [review]
Needs more work

File: engines/director/debugger.cpp [hunk 1]
```diff
@@ -61,4 +61,18 @@ bool Debugger::lingoCommandProcessor(const char *inputOrig) {
 	return true;
 }
 
+void Debugger::debugLogFile(Common::String logs) {
+	debugPrintf("%s", logs.c_str());
+	if (!g_director->_traceLogFile.empty()) {
+		Common::DumpFile out;
```
[the remark this answers is not in the document]
Comment (r41k0u) [reply engines/director/debugger.cpp:67]: That would present a problem.
What if the filename changes during the session, as in the test movie? In that case, a new log file is created, and both the logfiles exist. Closing the DumpFile object only once per session won't be able to handle change of log file name when the movie is running.

File: engines/director/lingo/lingo-the.cpp [hunk 2]
```diff
@@ -1130,7 +1132,10 @@ void Lingo::setTheEntity(int entity, Datum &id, int field, Datum &d) {
 		g_lingo->_traceLoad = d.asInt();
 		break;
 	case kTheTraceLogFile:
-		setTheEntitySTUB(kTheTraceLogFile);
+		if (d.asString().size())
+			g_director->_traceLogFile = ConfMan.get("path") + "/" + d.asString();
```
[the remark this answers is not in the document]
Comment (r41k0u) [reply engines/director/lingo/lingo-the.cpp]: I have used FSNode's isWriteable() to test the same now. Is that fine or I shall go by the seeking method?

File: engines/director/debugger.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (r41k0u) [reply engines/director/debugger.cpp]: Nevermind, I was abe to find a workaround which would do what you had advised and also solve the above problem

File: engines/director/lingo/lingo-the.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (sev-) [reply engines/director/lingo/lingo-the.cpp]: Yes, this would work.

**Comment by sev-** [comment]
THis has to be rebased and code formatting fixed.

**Comment by sev-** [comment]
Hello? @r41k0u

**Comment by sev-** [comment]
Merging, however, I will add after this a `warning()` if the log file is not writeable.

**Pull request merged**